app.js

```diff
@@ -160,15 +160,30 @@ async function shallowModeWarnings(opts) {
  */
 
 async function validateLogOnResume(opts) {
-  if (!opts || !opts.resume) {
+  const logFileExists = opts && opts.log && fs.existsSync(opts.log);
+  if (!opts || opts.mode === 'shallow') {
+    // No opts specified, defaults will be populated.
+    // In shallow mode log/resume are irrelevant and we'll have warned already.
     return true;
-  }
-  if (!opts.log) {
-    // This is the second place we check for the presence of the log option in conjunction with resume
-    // It has to be here for the API case
-    throw new BackupError('NoLogFileName', 'To resume a backup, a log file must be specified');
-  } else if (!fs.existsSync(opts.log)) {
-    throw new BackupError('LogDoesNotExist', 'To resume a backup, the log file must exist');
+  } else if (opts.resume) {
+    // Expecting to resume
+    if (!opts.log) {
+      // This is the second place we check for the presence of the log option in conjunction with resume
+      // It has to be here for the API case
+      throw new BackupError('NoLogFileName', 'To resume a backup, a log file must be specified');
+    } else if (!logFileExists) {
+      throw new BackupError('LogDoesNotExist', 'To resume a backup, the log file must exist');
+    }
+    if (opts.bufferSize) {
+      // Warn that the bufferSize is already fixed
+      console.warn('WARNING: the original backup "bufferSize" applies when resmuming a backup.');
```

Comment:
```suggestion
      console.warn('WARNING: the original backup "bufferSize" applies when resuming a backup.');
```

Reply:
I'd copied and pasted this error straight into the test too!
Thanks for spotting it.